pyupgrade.py

```diff
@@ -1227,12 +1259,43 @@ def _visit_simple(self, node):  # type: (NameOrAttr) -> None
             self.six_simple[_ast_to_offset(node)] = node
         self.generic_visit(node)
 
-    visit_Attribute = _visit_simple
+    visit_Attribute = visit_Name = _visit_simple
 
-    def visit_Name(self, node):  # type: (ast.Name) -> None
-        if node.id in self.OS_ERROR_ALIASES:
-            self.simple_ids[_ast_to_offset(node)] = node
-        self._visit_simple(node)
+    def visit_Try(self, node):  # type: (ast.Try) -> None
+        for handler in node.handlers:
+            htype = handler.type
+            if not htype:
+                continue
+            elif self._is_os_error_alias(htype):
+                assert isinstance(htype, (ast.Name, ast.Attribute))
+                self.os_error_alias_simple[_ast_to_offset(htype)] = htype
+            else:
+                assert isinstance(htype, ast.Tuple)
```

Comment:
```python
try:
    pass
except AssertionError:
    pass
```

```console
$ pyupgrade t.py  --py3-plus
Traceback (most recent call last):
  File "/home/asottile/workspace/pyupgrade/venv/bin/pyupgrade", line 11, in <module>
    load_entry_point('pyupgrade', 'console_scripts', 'pyupgrade')()
  File "/home/asottile/workspace/pyupgrade/pyupgrade.py", line 2062, in main
    ret |= fix_file(filename, args)
  File "/home/asottile/workspace/pyupgrade/pyupgrade.py", line 2031, in fix_file
    contents_text = _fix_py3_plus(contents_text)
  File "/home/asottile/workspace/pyupgrade/pyupgrade.py", line 1708, in _fix_py3_plus
    visitor.visit(ast_obj)
  File "/usr/lib/python3.6/ast.py", line 253, in visit
    return visitor(node)
  File "/home/asottile/workspace/pyupgrade/pyupgrade.py", line 1443, in generic_visit
    super(FindPy3Plus, self).generic_visit(node)
  File "/usr/lib/python3.6/ast.py", line 261, in generic_visit
    self.visit(item)
  File "/usr/lib/python3.6/ast.py", line 253, in visit
    return visitor(node)
  File "/home/asottile/workspace/pyupgrade/pyupgrade.py", line 1273, in visit_Try
    assert isinstance(htype, ast.Tuple)
AssertionError
```

Reply:
also

```python
try:
    pass
except [AssertionError] as e:
    pass
```

```python
try:
    pass
except {AssertionError} as e:
    pass
```

```python
def hello_world():
    return AssertionError

try:
    pass
except hello_world() as e:
    pass
```

Reply:
makes sense 🤦‍♂ this assertion was to work around coverage

```diff
-            else:
-                assert isinstance(htype, ast.Tuple)
+            elif isinstance(htype, ast.Tuple):
```

![image](https://user-images.githubusercontent.com/252209/61141658-0b1fd600-a49c-11e9-9d19-6387ffd12501.png)

I'll work around this another way

Reply:
Adding that test case and restructuring the code a bit did the trick 👍

Reply:
just means you need more tests :P